storefront.rb

```diff
@@ -3,11 +3,123 @@
+puts "\n"
+# How much was spent on books?
+spent = Item.joins(:orders).where("category LIKE '%books%'").sum("price*quantity")
+puts "$#{spent} was spent on books."
+
+######### Adventure Mode
+
+# What item was ordered most often? Grossed the most money?
+ most_order = Order.joins(:item).order("sum_orders_quantity desc").group(:title).sum("orders.quantity").first
+ puts "#{most_order.first} was ordered the most."
+
+ gross = Order.joins(:item).order("sum_orders_quantity_all_items_price desc").group(:title).sum("orders.quantity * items.price").first
+ puts "#{gross.first} grossed the most money."
+
+# What user spent the most?
+
+baller = Order.joins(:user, :item).order("sum_orders_quantity_all_items_price desc").group("users.id").sum("orders.quantity * items.price").first
+baller_name = User.where("id = ?", baller.first).first
+puts "#{baller_name.first_name} #{baller_name.last_name} spent the most money.\n\n"
+## There has to be a better way to do this
+
+# What were the top 3 highest grossing categories?
+
+gc = Item.joins(:orders).order("sum_orders_quantity_all_items_price").reverse_order.group("items.category").sum("orders.quantity * items.price").first(3)
+puts "The top 3 grossing categories were:\n #{gc[0].first} with $#{gc[0].last},\n #{gc[1].first} with $#{gc[1].last},\n and #{gc[2].first} with $#{gc[2].last}."
+## Note: Research how to do adventure mode better
+
+# Simulate buying an item by inserting a User from command line input (ask the user for their information) and an Order for that User (have them pick what they'd like to order and other needed order information).
+prompt = TTY::Prompt.new
+
+if prompt.yes?("Would you like to buy an item?")
+  first_name = prompt.ask("What is your first name?")
+  last_name = prompt.ask("What is your last name?")
+  email = prompt.ask("What is your email?")
+  user = User.create(first_name: first_name, last_name: last_name, email: email)
+  puts "We added #{user.first_name} #{user.last_name} to our database! Thank you!"
+  all_items = Item.order(:price).group_by{|i| i[:title]}
+  item = prompt.select("What item would you like to buy?", all_items).first
+  amount = prompt.ask("How many would you like to buy?")
+  buy = Order.create(user: user, item: item, quantity: amount)
+  puts "You have bought #{buy.quantity} of #{item.title}'s"
+else
+  puts "Have a nice day!"
+end
+
+#Putting this out of order so when the script is run it outputs all of the information and then does the prompts
+
+
+###### Epic Mode
+
+# unfinished epic mode prompt
+#Can i put a condition where if the email doesn't exist return something?
+#This only works when I'm returning it as an array/hash
+#I could add a Y/N confirmation loop to check ordered item
+
+if prompt.yes?("Would you like to leave a review?")
+  find_email = prompt.ask("What is your email used when buying an item?")
+  find_id = User.where("email = ?", find_email).first.id
+  find_item = Order.where("user_id = ?", find_id).first
```

Comment:
Would rather give the user the option of selecting which order item they'd like to review.